.github/workflows/ci.yml

```diff
@@ -5,46 +5,49 @@ on:
   pull_request:
 
 env:
-  CONSUL_VERSION: 1.6.10
-
+  DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
+  DOTNET_CLI_TELEMETRY_OPTOUT: true
+    
 jobs:
-
-  # Enforces the consistency of code formatting using `.editorconfig` and the `dotnet-format` tool.
-  check-format:
+  Consul_AspNetCore:
     if: github.event_name == 'push' || github.event.pull_request.head.repo.id != github.event.pull_request.base.repo.id
-    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, windows-2019, macos-10.15]
+        framework: [netcoreapp3.1]
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
     steps:
       - name: Checkout
         uses: actions/checkout@v2
-      - name: Setup .NET Core SDK
+      - name: Setup .NET Core SDK v3.1.x
         uses: actions/setup-dotnet@v1
         with:
-          dotnet-version: 3.1.404
-      - name: Restore tools
-        run: dotnet tool restore
-      - name: Check format
-        run: dotnet format --check
-
-  # Build and test on all major platforms.
-  # Repack to avoid Newtonsoft.Json dependency in .NET Framework (This is to
-  # prevent issues with mismatched JSON.NET versions in user projects).
-  # Create a NuGet package on Windows only, as we build for .NET Framework 4.6.1
-  # and for .NET Standard 2.0.
-  build:
+          dotnet-version: 3.1.x
+      - name: Build
+        run: dotnet build Consul.AspNetCore.Test --configuration=Release -p:TreatWarningsAsErrors=true --framework=${{ matrix.framework }}
+      - name: Run tests
+        shell: bash
+        run: dotnet test Consul.AspNetCore.Test --configuration=Release --logger "GitHubActions;report-warnings=false" --no-build -v=Normal --framework=${{ matrix.framework }}
+  Consul:
     if: github.event_name == 'push' || github.event.pull_request.head.repo.id != github.event.pull_request.base.repo.id
     strategy:
-      matrix:
+      matrix:      
+        consul: [1.6.10, 1.7.10, 1.8.6, 1.9.0]
+        framework: [net461, netcoreapp2.1, netcoreapp3.1, net5.0]
         os: [ubuntu-18.04, windows-2019, macos-10.15]
       fail-fast: false
     runs-on: ${{ matrix.os }}
     steps:
       - name: Checkout
         uses: actions/checkout@v2
```

Comment:
I think we should add a step to install the latest .NET 5.0.x SDK to have consistency between operating systems, otherwise we will be at the mercy of the runners being upgraded at their own pace.

Reply:
It is already https://github.com/G-Research/consuldotnet/blob/c1bb616cac7c508ee878f8aeab2ed1eb52040bbd/.github/workflows/ci.yml#L59

Reply:
sorry, my bad!